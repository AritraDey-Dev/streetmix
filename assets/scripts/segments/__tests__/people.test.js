/* eslint-env jest */
import { pickObjectsFromPool } from '../people'
import PEOPLE from '../people.json'

// The unit test for `pickObjectsFromPool()` is designed to ensure that it
// returns the same values as previous behaviour, because we're heavily
// refactoring how this works, and want to avoid regressions. We will need to
// continue making sure that we return similar output against future
// refactoring -- for instance, when transitioning from imperial to metric
// units.

// However! The output of this function is not intended to be guaranteed
// consistent across all releases, because it can change for any reason -- for
// example, just adding more people to people.json will result in different
// output. As a future improvement, we can either make this test more general,
// or roll it up into an integration test.
describe('scatter objects in segments', () => {
  it('picks people at normal density', () => {
    const [people, startLeft] = pickObjectsFromPool(
      PEOPLE,
      28.66667,
      936877893,
      0,
      5,
      4,
      -0.5
    )

    expect(people).toMatchSnapshot()
    expect(startLeft).toEqual(3.7987234759633886)
  })

  it('picks people at sparse density', () => {
    const [people, startLeft] = pickObjectsFromPool(
      PEOPLE,
      28.66667,
      936877893,
      3.5,
      11.8333,
      4,
      -0.5
    )

    expect(people).toMatchSnapshot()
    expect(startLeft).toEqual(4.846658384686)
  })

  it('picks people at dense density', () => {
    const [people, startLeft] = pickObjectsFromPool(
      PEOPLE,
      28.66667,
      936877893,
      0,
      1.5,
      4,
      -0.5
    )

    expect(people).toMatchSnapshot()
    expect(startLeft).toEqual(3.0462522570605604)
  })
})
