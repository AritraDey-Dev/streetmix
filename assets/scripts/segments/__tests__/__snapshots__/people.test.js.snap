// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scatter objects in segments picks people at dense density 1`] = `
Array [
  Object {
    "id": "people-25",
    "left": 0,
    "width": 2,
  },
  Object {
    "disallowFirst": true,
    "id": "johnny-01",
    "left": 2.74073930948676,
    "name": "Johnny",
    "width": 3,
  },
  Object {
    "id": "people-24",
    "left": 5.615210042353209,
    "width": 4,
  },
  Object {
    "id": "people-25",
    "left": 9.860021493652413,
    "width": 2,
  },
  Object {
    "id": "people-27",
    "left": 12.368637363837975,
    "width": 2,
  },
  Object {
    "id": "people-13",
    "left": 14.270766914421968,
    "width": 2,
  },
  Object {
    "id": "people-26",
    "left": 16.98367551508844,
    "width": 3,
  },
  Object {
    "id": "people-16",
    "left": 20.61886035144698,
    "name": "Lou",
    "width": 3,
  },
  Object {
    "disallowFirst": true,
    "id": "johnny-01",
    "left": 24.074165485878876,
    "name": "Johnny",
    "width": 3,
  },
]
`;

exports[`scatter objects in segments picks people at normal density 1`] = `
Array [
  Object {
    "id": "people-25",
    "left": 0,
    "width": 2,
  },
  Object {
    "disallowFirst": true,
    "id": "johnny-01",
    "left": 5.6357976982892,
    "name": "Johnny",
    "width": 3,
  },
  Object {
    "id": "people-24",
    "left": 9.384033474510698,
    "width": 4,
  },
  Object {
    "id": "people-25",
    "left": 15.36673831217471,
    "width": 2,
  },
  Object {
    "id": "people-27",
    "left": 20.228791212793247,
    "width": 2,
  },
  Object {
    "id": "people-13",
    "left": 23.069223048073223,
    "width": 2,
  },
]
`;

exports[`scatter objects in segments picks people at sparse density 1`] = `
Array [
  Object {
    "id": "people-25",
    "left": 0,
    "width": 2,
  },
  Object {
    "disallowFirst": true,
    "id": "johnny-01",
    "left": 11.892968591830677,
    "name": "Johnny",
    "width": 3,
  },
  Object {
    "id": "people-24",
    "left": 19.973353230628,
    "width": 4,
  },
]
`;
